### PR TITLE
Add reqwest as a feature

### PR DESCRIPTION
**Pull request by 4JX**
Still does not work on windows due to `ifaces`, but one step closer to doing so. (Seems the [repo](https://github.com/misuzu/rust-interfaces) its forked from has working Windows support, lemme know if I can try to fit switching to that in this PR)

Is that crate only used for detecting network related changes?

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -44,3 +49,8 @@ version = "1.17.0"
 features = ["fs", "io-util", "rt", "sync", "time"]
 
 [features]
+default = ["isahc"]
```
Comment (mmstick) [inline Cargo.toml]: Features should be additive. There shouldn't be any default features.

**Review by mmstick — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (4JX) [reply Cargo.toml]: The lack of a default would mean the crate won't compile without an explicit feature, is that OK or did you mean to always have `isahc` support not locked behind a feature?

Related to the latter, when two crates used as dependencies each use one of the features the current code won't compile. The only solution I can think of is to have an abstracted `Client`, `Request`, etc that has functions to convert into the crate specific structs while keeping track of the "backend" used. Similar to how `reqwest` has that for openssl/rusttls.

Comment (mmstick) [reply Cargo.toml]: It's a best practice for handling features in Cargo crates. Features should always be additive, and default features should never conflict with extra features. It should be a compiler error.

Comment (4JX) [reply Cargo.toml]: Should be addressed with the latest commit

File: Cargo.toml [hunk 2]
```diff
@@ -44,3 +49,4 @@ version = "1.17.0"
 features = ["fs", "io-util", "rt", "sync", "time"]
 
 [features]
+reqwest = ["dep:reqwest"]
```
Comment (mmstick) [inline Cargo.toml:54]: Missing an isahc feature

File: src/get.rs [hunk 3]
```diff
@@ -42,95 +42,158 @@ pub(crate) async fn get<Data: Send + Sync + 'static>(
...
-                .await
-                .map_err(Error::from)
-        };
-
-        let initial_response = crate::utils::timed_interrupt(Duration::from_secs(10), req).await?;
-
-        if initial_response.status() == StatusCode::NOT_MODIFIED {
-            return Ok::<_, crate::Error>((dest, file));
+        match &fetcher.client {
+            Client::Isahc(client) => {
```
Comment (mmstick) [inline src/get.rs:55]: isahc shouldn't be compiled or linked when reqwest is enabled

**Review by mmstick — CHANGES_REQUESTED** [review]

File: src/get.rs (hunk 3 above)
Comment (4JX) [reply src/get.rs:55]: If both features are enabled, both should still be available right?

Comment (mmstick) [reply src/get.rs:55]: One or the other, but never both at the same time.

**Review by mmstick — APPROVED** [review]

**Pull request merged**